lib/src/utils/commands_extension.dart

```diff
@@ -21,29 +21,38 @@ import 'dart:convert';
 
 import 'package:matrix/matrix.dart';
 
+/// callback taking [CommandArgs] as input and a [StringBuffer] as standard output
+/// optionally returns an event ID as in the [Room.sendEvent] syntax.
+/// a [CommandException] should be thrown if the specified arguments are considered invalid
+typedef CommandExecutionCallback = FutureOr<String?> Function(
+  CommandArgs,
+  StringBuffer? stdout,
```

Comment:
I don't see the `stdout` used in any of the default commands. Even though having an output buffer sounds like an important thing to have, I doubt how useful it'll be. I just can't think of a really good use-case for it right now, could you pls tell me a good way it can be used (to output useful data) with the default commands?

Reply:
Yes, it's true, I didn't change the behavior of the default commands. Mostly, since they do not really have anything to output because they are pretty static commands without any kind of user interaction.

I personally use the stdout to modify the message text field : I e.g. implemented a `/poll` command assisting in sending composing Element polls, I added a `/search` command returning a protobuf content stream. I intentionally didn't specify any high-level format to use for stdout since I can imagine infinite different use cases how clients would like to interact with the output of commands that is not an eventId - that's why ; a simple `StreamBuffer` and nothing further.

Reply:
One could consider adding JSON output of the `/dm` and `/create` commands containing the room ID though. If you like, I will add this.

Reply:
@coder-with-a-bushido See latest comment. I added a helper class for command output and now add the room IDs of created/searched rooms to the stdout in the default commands.